File: Лаба2/Лаба.cpp
/*
ЗАДАНИЕ 1.
Даны длины ребер a, b, c прямоугольного параллелепипеда.
Найти его объем V = a·b·c и площадь поверхности S = 2·(a·b + b·c + a·c).

ЗАДАНИЕ 2
Учитывая массив nums целых чисел, 
вернуть, сколько из них содержат нечетное количество цифр.

ЗАДАНИЕ 3
Для каждого трёхзначного числа определить является ли оно палиндромом 
(Читается одинаково с любой стороны, например 101, 222 и т.д.).
*/
#include <iostream>
#include <vector>
using namespace std;

void Task1()
{
	int a, b, c;
	cout << "Enter the length:";
		cin >> a;
	cout << "\nEnter the width:";
		cin >> b;
	cout << "\nEnter the height:";
		cin >> c;
	int V, S;
	S = 2 * (a * b + b * c + a * c);
	cout << "S = " << S << endl;
	V = a * b * c;
	cout << "V = " << V << endl;
}

void Task2()
{
	int n, s;        //n - кол-во входящих чисел, s - кол-во подходящих чисел
	s = 0;
	cout << "Kol-vo chisel v viborke = "; cin >> n;
	vector <int> nums(n, 0);       //инициализация массива

	for (int i = 1; i <= n; i++) {
		nums[i] = rand() % 9999;         //рандом значений
		cout << nums[i] << " ";             //вывод заполненного массива
	}

	cout << "\n";

	for (int i = 1; i <= n; i++) {         //перебор значений
		int o = nums[i];      //значение из вектора
		int k = 0;                  //кол-во цифр
		if (o != 0) {                //проверка на 0
			while (o != 0) {                      //подсчёт цифр
				o = o / 10;
				k++;
			}
			if (k % 2 == 1)
				s++;
		}
	}

	cout << s;
	
}

void Task3()
{
	while (true)
	{
		int num;
		cout << "Enter The number:";
		cin >> num;
		if (num >= 100 && num < 1000) { // Провека, является ли число трёхзначным или нет
			if (num / 100 == num % 10)// Проверка, является ли число полиндромом
				cout << "Yes" << endl;
			else cout << "No" << endl;
		}
		else
			cout << "No" << endl;
		num = 0;
		bool res;
		cout << "Continue or not(1/0):";
		cin >> res;
		if (res == 0)
			break;
	}
}


int main()
{
	int task;
	cout << "Select a task:";
	cin >> task;
	if (task == 1) {
		cout << "Task 1:" << endl;
		Task1();
	}
	if (task == 2) {
		cout << "\nTask 2:" << endl;
		Task2();
	}
	if (task == 3) {
		cout << "\nTask 3:" << endl;
		Task3();
	}
	else
	{
		cout << "Invalid value";
	}
	return 0;
}

File: 2-й семестр/Лаба2/Lab2 питон/Лаба.py
import math
import random
class Globals:

    @staticmethod
    def Task1():
        a = None
        b = None
        c = None
        print("Enter the length:", end = '')
        a = input
        print("\nEnter the width:", end = '')
        b = input
        print("\nEnter the height:", end = '')
        c = input
        V = None
        S = None
        S = 2 * (a * b + b * c + a * c)
        print("S = ", end = '')
        print(S, end = '')
        print("\n", end = '')
        V = a * b * c
        print("V = ", end = '')
        print(V, end = '')
        print("\n", end = '')

    @staticmethod
    def Task2():
        n = None #n - кол-во входящих чисел, s - кол-во подходящих чисел
        s = None
        s = 0
        print("Enter the number of items in the selection: ", end = '')
        n=input
        nums = [0 for _ in range(n)] #инициализация массива

        for i in range(1, n + 1):
            nums[i] = random.random() #рандом значений
            print(nums[i], end = '')
            print(" ", end = '')

        print("\n", end = '')

        for i in range(1, n + 1):
            o = nums[i] #значение из вектора
            k = 0 #кол-во цифр
            if o != 0:
                while o != 0:
                    o = math.trunc(o / float(10))
                    k += 1
                if math.fmod(k, 2) == 1:
                    s += 1

        print(s, end = '')


    @staticmethod
    def Task3():
        while True:
            num = None
            print("Enter The number:", end = '')
            num = input
            if num >= 100 and num < 1000:
                if math.trunc(num / float(100)) == math.fmod(num, 10): # Проверка, является ли число полиндромом
                    print("Yes", end = '')
                    print("\n", end = '')
                else:
                    print("No", end = '')
                    print("\n", end = '')
            else:
                print("No", end = '')
                print("\n", end = '')
            num = 0
            res = None
            print("Continue or not(1/0):", end = '')
            res = input
            if res == False:
                break
def main():
    task = None
    print("Select a task:", end = '')
    task = input
    if task == 1:
        print("Task 1:", end = '')
        print("\n", end = '')
        Globals.Task1()
    if task == 2:
        print("\nTask 2:", end = '')
        print("\n", end = '')
        Globals.Task2()
    if task == 3:
        print("\nTask 3:", end = '')
        print("\n", end = '')
        Globals.Task3()
    else:
        print("Invalid value", end = '')
